biofefi/machine_learning/data.py

```diff
@@ -34,20 +35,41 @@ def __init__(
         self._normalization = normalization
         self._numerical_cols = "all"
         self._n_bootstraps = n_bootstraps
+        self._create_synthetic_data = False
 
     def _load_data(self) -> Tuple[pd.DataFrame, pd.DataFrame]:
         """
-        Load data from a csv file
+        Load data from a csv file or
+        generate synthetic data
 
-        Returns
-        -------
-        Tuple[pd.DataFrame, pd.DataFrame]
-            The training data (X) and the targets (y)
+        Returns:
+            if synthetic data is used:
+                - Tuple: The training data and prediction targets
+                which are created artificially.
+
+            if csv file is used:
+                - Tuple: The training data and prediction targets
+                loaded from a csv file.
```

Comment:
⛏️ You don't need to document whether it's returning synthetic or real data. This class is just loading a csv file as far as it's concerned